core/src/main/java/nl/inl/blacklab/index/config/DocIndexerXPath.java

```diff
@@ -571,6 +577,59 @@ protected void processAnnotation(ConfigAnnotation annotation, List<Integer> inde
         }
     }
 
+    /**
+     * if needed replace ' with &apos; in the value of a captureValuePath and make sure that when the value is 
+     * actually retrieved the replacement is reverted to get the correct value in the index.
+     * @param value
+     * @param annotation
+     * @return 
+     */
+    private String processCaptureValuePath(String valuePath, int i, String value, ConfigAnnotation annotation) {
+        /*
+        The value is injected in an xpath which may use quoting in different ways.
+        The value may contain quotes that conflict with the quoting used in the xpath.
+        We only deal with the situation where the variable is quoted like this: '$i'
+        */
+        if (valuePath.contains("'$"+i+"'") && value.contains("'")) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("replacing ' with &apos; in " + value + " before injecting in " + valuePath);
+            }
+            value = value.replace("'", "&apos;");
+            /**
+             * now we need to finally revert the &apos; replacement, if needed we add a process for this
+             */
+            if (annotation.getProcess().isEmpty()) {
+                annotation.getProcess().add(revertEscapeQuote());
```

Comment:
Why is this required? This works on the value, not the escaped xpath expression. Values are from the parsed XML document and shouldn't contain &apos; (or any other entity references for that matter) anymore, right?

Reply:
becasue an xpath can do concat('$1'..... where $1 now contains &apos;

Reply:
Okay, I understand.